web/src/features/auth/components/MoreCast2AuthWrapper.tsx

```diff
@@ -24,23 +25,31 @@ const MoreCast2AuthWrapper = ({ children }: Props) => {
       if (TEST_AUTH || window.Cypress) {
         dispatch(wf1Authenticate('test token'))
         setRenderChildren(true)
-      }
+      } else {
+        if (!isAuthenticatedForecaster) {
+          setRenderChildren(true)
+        }
 
-      if (!isAuthenticatedForecaster) {
-        setRenderChildren(true)
-      }
+        const lastLoginString = localStorage.getItem('last_morecast_login') ?? '0'
+        const lastLogin = parseInt(lastLoginString)
+        const now = DateTime.now().toUnixInteger()
 
-      if (!window.location.href?.includes('access_token') && isAuthenticatedForecaster) {
-        window.location.href = `${WF1_AUTH_URL}&redirect_uri=${window.location.href}`
-      }
+        // Force a redirect to WF1 authentication if there is no access_token or the last time Morecast
+        // was logged into was more than 60 minutes ago in order to handle a user reloading a page with an
+        // old access token
+        if (!window.location.href?.includes('access_token') || now - lastLogin > 30) {
+          window.location.href = `${WF1_AUTH_URL}&redirect_uri=${window.location.href}`
```

Comment:
I think we need to strip off the `access_token` parameter here before redirecting.

Reply:
Might be simpler to just do something like:

```
window.location.href = window.location.origin + "/" + window.location.pathName
```

Reply:
Actually I read this a little wrong I think it should be:

```
window.location.href = `${WF1_AUTH_URL}&redirect_uri=${window.location.origin + "/" + window.location.pathName}`
```

So that we don't keep the old token. Or does this work without that? I tried manually setting the local storage timestamp manually and I was getting the same token, but maybe that's just WF1 auth.

Reply:
The WF1 endpoint strips off the old token, but better to be explicit.